### PR TITLE
Improve bee-message test coverage

### PR DESCRIPTION
**Pull request by Adam-Gleave**
# Description of change

Increased test coverage for the `bee-message` crate.

## Type of change

- Enhancement

## How the change has been tested

Tested locally to ensure all new cases pass. Unfortunately the coverage workflow is faulty right now, so no exact figures on coverage after these changes.

## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

File: bee-message/tests/ed25519_signature.rs [hunk 1]
```diff
@@ -6,7 +6,7 @@ use bee_message::prelude::*;
 
 use core::convert::TryInto;
 
-const ED25519_PULIC_KEY: &str = "1DA5DDD11BA3F961ACAB68FAFEE3177D039875EAA94AC5FDBFF8B53F0C50BFB9";
+const ED25519_PUBLIC_KEY: &str = "1DA5DDD11BA3F961ACAB68FAFEE3177D039875EAA94AC5FDBFF8B53F0C50BFB9";
```
Comment (thibault-martinez) [inline bee-message/tests/ed25519_signature.rs]: I know it was already there but could you make it lowercase for consistency ?

File: bee-message/tests/milestone.rs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    assert_eq!(
+        format!("{:?}", milestone),
+        "Milestone { message_id: MessageId(9e23e9fccb816af4ad355c27d904b6a6e88618e0bed1b640df3d4c19f4579bc9), timestamp: 0 }",
+    );
+}
+
+#[test]
+fn packed_len() {
+    let milestone = rand_milestone();
+    assert_eq!(milestone.packed_len(), milestone.pack_new().len());
```
Comment (thibault-martinez) [inline bee-message/tests/milestone.rs:25]: I would also add a check with the actual expected hardcoded length. We don't have the guarantee both packed_len and pack_new().len() aren't faulty the same way atm

File: bee-message/tests/transaction_payload.rs [hunk 3]
```diff
@@ -8,7 +8,7 @@ use core::convert::TryInto;
 
 const TRANSACTION_ID: &str = "52fdfc072182654f163f5f0f9a621d729566c74d10037c4d7bbb0407d1e2c649";
 const ED25519_ADDRESS: &str = "52fdfc072182654f163f5f0f9a621d729566c74d10037c4d7bbb0407d1e2c649";
-const ED25519_PULIC_KEY: &str = "1DA5DDD11BA3F961ACAB68FAFEE3177D039875EAA94AC5FDBFF8B53F0C50BFB9";
+const ED25519_PUBLIC_KEY: &str = "1DA5DDD11BA3F961ACAB68FAFEE3177D039875EAA94AC5FDBFF8B53F0C50BFB9";
```
Comment (thibault-martinez) [inline bee-message/tests/transaction_payload.rs]: Lowercase

**Review by thibault-martinez — CHANGES_REQUESTED** [review]
Clippy + tests

File: bee-message/tests/ed25519_signature.rs (hunk 1 above)
Comment (Adam-Gleave) [reply bee-message/tests/ed25519_signature.rs]: Sure, didn't notice that!

File: bee-message/tests/milestone.rs (hunk 2 above)
Comment (Adam-Gleave) [reply bee-message/tests/milestone.rs:25]: Yep, will do.

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**